File: Java/Bikes/src/MountainBike.java
class MountainBike extends Bicycle {

    // the MountainBike subclass adds one more field
    private int seatHeight;

    // the MountainBike subclass has one constructor
    public MountainBike(int gear, int speed, int seatHeight) {
        super(gear, speed); // invoking base-class(Bicycle) constructor
        this.seatHeight = seatHeight;
    }

    public void setHeight(int seatHeight) {
        this.seatHeight = seatHeight;
    }

    @Override
    public String toString() {
        return (super.toString() + "\nseat height is "
                + seatHeight);
    }

    @Override
    public boolean equals(Object obj) {

        if (obj instanceof MountainBike) {
            MountainBike mb2 = (MountainBike) obj;
            return super.equals(mb2) && mb2.seatHeight == this.seatHeight;
        } else
            return false;
    }
}

File: Python/Bikes/MountainBike.py
from Bicycle import Bicycle

class MountainBike(Bicycle):
    
    __seatHeight = None


    def __init__(self,  gear,  speed,  seatHeight):
        super().__init__( gear, speed)
        self.__seatHeight = seatHeight
    
    def setHeight(self, seatHeight):
        self.__seatHeight = seatHeight
 
    def __str__(self):
        return super().__str__() + "\nseat height is "\
                + str(self.__seatHeight)

    def __eq__(self,bk2):
        return super().__eq__(bk2) and self.__seatHeight == bk2.__seatHeight
